### PR TITLE
feat(yeti): Elevation profile: new tool and improvements

### PR DESCRIPTION
**Pull request by iamvdo**
Hello,

This PRs adds **some improvements** to the elevation profile:

- dashed lines between line chunks
- altitude popup on hover
- perf improvements

It also adds a **new tool to generate elevations data**, based on external API. And fixes some bugs here and there.

Here is a screen video of the result

https://user-images.githubusercontent.com/1260529/227963446-7eebb4fc-fce8-4fa8-b367-51023e4b7822.mp4

_
Thanks for review.

### CODE REVIEW

File: src/components/yeti/ElevationProfile.vue [hunk 1]
```diff
@@ -19,33 +76,181 @@ export default {
...
+      // start loading
+      this.loading = true;
+      // and defer (next cycle)
+      setTimeout(() => {
+        if (this.override) {
+          // when force elevation, start by remove all elevations
+          this.removeAllElevations();
+        }
+
+        // interpolateValue is 5 meters min, or 0 is not checked
```
Comment (brunobesson) [inline src/components/yeti/ElevationProfile.vue:199]:
```suggestion
        // interpolateValue is 5 meters min, or 0 if not checked
```

File: src/components/yeti/ElevationProfile.vue [hunk 2]
```diff
@@ -268,6 +488,17 @@ export default {
         .attr('class', 'area')
         .attr('d', this.area);
 
+      // build areas line (dashed)
+      let areasDataForLines = JSON.parse(JSON.stringify(this.areasData));
```
Comment (brunobesson) [inline src/components/yeti/ElevationProfile.vue:492]: we can use [struturedClone](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) instead

**Review by brunobesson — APPROVED** [review]
(sorry, only time for a quick and fast review, I trust you)

File: src/components/yeti/ElevationProfile.vue [hunk 3]
```diff
@@ -19,33 +76,181 @@ export default {
...
-
   beforeDestroy() {
     this.resizeObserver?.unobserve(this.$refs.graph);
   },
-
   methods: {
+    onCalculateElevations() {
+      // deal with errors first
+      if (this.validSimplifyTolerance) {
+        return window.alert(this.errors.actionDisabled);
```
[the remark this answers is not in the document]
Comment (iamvdo) [reply src/components/yeti/ElevationProfile.vue:180]: Sorry, didn't notice the use of toasts in the app.
But, will let this as-is for now. We need to test how toasts are working (options, durations, etc.), and discuss all types of errors from the whole YETI app from our side beforehand.

**Pull request merged**